Application/EdFi.Ods.Api/Helpers/AssemblyLoaderHelper.cs

```diff
@@ -146,6 +147,62 @@ public static IEnumerable<string> FindPluginAssemblies(string pluginFolder, bool
                 yield break;
             }
 
+            var apiModelFiles = Directory.GetFiles(pluginFolder, "ApiModel-EXTENSION.json", SearchOption.AllDirectories);
+
+            var physicalNames = new List<KeyValuePair<string, string>>();
+
+            Array.Sort(apiModelFiles, StringComparer.InvariantCulture);
+
+            foreach (var apiModelFilePath in apiModelFiles)
+            {
+                var myJsonString = File.ReadAllText(apiModelFilePath);
+
+                var myJsonObject = JObject.Parse(myJsonString);
+
+                JToken physicalNameJToken = myJsonObject.SelectToken("$.schemaDefinition.physicalName");
+
+                if (physicalNameJToken != null)
+                {
+                    string[] foldernames = apiModelFilePath.Split(Path.DirectorySeparatorChar);
+                    foreach (string folder in foldernames)
+                    {
+                        if (folder.Contains("Extensions"))
+                        {
+                            string key = physicalNameJToken.Value<string>();
+                            var element = new KeyValuePair<string, string>(key, folder);
+                            physicalNames.Add(element);
+                        }
+                    }
+                } 
+            }
+
+            string duplicatePhysicalName = "";
+            string duplicatePluginFolder = "";
+            bool IsDuplicateExist = false;
+
+            foreach (var physicalName in physicalNames)
+            {
+                if (physicalName.Key.EqualsIgnoreCase(duplicatePhysicalName))
+                {
+                    _logger.Error($"found duplicate extension schema name '{physicalName.Key}' in plugin folder." +
+                        $" You will be able to deploy only one of the following plugins '{physicalName.Value}' and '{duplicatePluginFolder}' folder name." +
+                        $" Please remove the conflicting plugins and retry");
+                    IsDuplicateExist = true;
+                    duplicatePhysicalName = "";
+                    duplicatePluginFolder = "";
+                }
+                else
+                {
+                    duplicatePhysicalName = physicalName.Key;
+                    duplicatePluginFolder = physicalName.Value;
+                }
+            }
+
+            if(IsDuplicateExist)
+            {
+                throw new Exception();

```

Comment:
Now that we've got this all laid out and working I could see the argument for having a message here.